lib/components/normal-components/Switch.ts

```diff
@@ -0,0 +1,49 @@
+import { switchProps } from "@tscircuit/props"
+import { NormalComponent } from "../base-components/NormalComponent/NormalComponent"
+import type { BaseSymbolName } from "lib/utils/constants"
+
+export class Switch extends NormalComponent<typeof switchProps> {
+  private _getSwitchType(): "spst" | "spdt" | "dpst" | "dpdt" {
+    const { spst, spdt, dpst, dpdt, type } = this._parsedProps ?? {}
+
+    if (dpdt) return "dpdt"
+    if (spst) return "spst"
+    if (spdt) return "spdt"
+    if (dpst) return "dpst"
+    return type ?? "spst"
+  }
+
+  get config() {
+    const switchType = this._getSwitchType()
+
+    const baseSymbolName: Record<string, string> = {
```

Comment:
bad var name, also unnecessary type def

```suggestion
    const baseSymbolNameMap = {
```